spec.md

```diff
@@ -1495,6 +1626,112 @@ For success responses, the following fields are defined:
 }
 ```
 
+## Adapting Service Binding Credentials
+
+If the endpoints of a Service Instance are from an Application point of view
+different to the endpoints provided in the binding credentials, then the
+binding credentials MUST be adapted before the Platform provides them to
+the Application.
+
+The Platform requests the corrected credentials from the Service Broker by
+providing the original credentials and a table that maps the original
+endpoints to the new endpoints.
+
+### Request
+
+##### Route
+
+`POST /v2/service_instances/:instance_id/service_bindings/:binding_id/adapt_credentials`
+
+`:instance_id` MUST be the ID of a previously provisioned Service Instance.
+
+`:binding_id` MUST be the ID of a previously provisioned Service Binding for that
+instance.
+
+#### Headers
+
+The following HTTP Headers are defined for this operation:
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### Body
+| Request Field | Type | Description |
+| --- | --- | --- |
+| credentials* | object | The free-form hash of credentials that the Platform has received from the Service Broker. |
+| endpoint_mappings* | array of [EndpointMapping](#endpoint-mapping-object) objects | A list of endpoint mapping. |
+
+\* Headers with an asterisk are REQUIRED.
+
+##### Endpoint Mapping Object
+
+| Request Field | Type | Description |
+| --- | --- | --- |
+| source* | [Endpoint](#endpoint-object) object | The endpoint on the service side. |
+| target* | [Endpoint](#endpoint-object) object | The mapped endpoint on the application side. |
+
+\* Headers with an asterisk are REQUIRED.
+
+```
+{
+  "credentials": {
+    "uri": "mysql://mysqluser:pass@mysqlhost:3306/dbname",
+    "username": "mysqluser",
+    "password": "pass",
+    "host": "mysqlhost",
+    "port": 3306,
+    "database": "dbname"
+  },
+  "endpoint_mappings": [
+    {
+      "source": {
+        "host": "mysqlhost",
+        "port": 3306
+      },
+      "target": {
+        "host": "appnethost",
+        "port": 9876
+      }
+    }
+  ]
+}
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | The expected response body is below. |
+| 404 Not Found | MUST be returned if the Service Binding does not exist or if a binding operation is still in progress. |
+
+Responses with any other status code MUST be interpreted as a failure and the
+Platform MUST continue to remember the Service Binding.
+
+##### Body
+
+For success responses, the following fields are defined:
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| credentials | object | The adapted free-form hash of the credentials. |
+
+```
+{
+  "credentials": {
+    "uri": "mysql://mysqluser:pass@appnethost:9876/dbname",
+    "username": "mysqluser",
+    "password": "pass",
+    "host": "appnethost",
+    "port": 9876,
+    "database": "dbname"
+  }
+}
+```
+
+
 ## Unbinding
```

Comment:
Unbinding operations should also contain enough information in the request and response to undo any network operations which occurred during the bind operations.